core-blocks/block/edit-panel/index.js

```diff
@@ -22,9 +22,29 @@ class SharedBlockEditPanel extends Component {
 
 		this.titleField = createRef();
 		this.editButton = createRef();
+		this.editForm = createRef();
 		this.handleFormSubmit = this.handleFormSubmit.bind( this );
 		this.handleTitleChange = this.handleTitleChange.bind( this );
 		this.handleTitleKeyDown = this.handleTitleKeyDown.bind( this );
+		this.handleClickOutside = this.handleClickOutside.bind( this );
+	}
+
+	componentDidMount() {
+		document.addEventListener( 'click', this.handleClickOutside, true );
+	}
+
+	componentWillUnmount() {
+		document.removeEventListener( 'click', this.handleClickOutside, true );
+	}
+
+	handleClickOutside( event ) {
+		if ( ! this.editForm.current ) {
```

Comment:
When will this be the case?

Reply:
The edit form is a separate component and mounted only when editing, If you "click outside" when _not_ editing, then the returned DOM element is `null`.

Reply:
But isn't the callback only bound to the `click` event when it is in-fact mounted?